src/python/pyhbac.c

```diff
@@ -1947,44 +1946,90 @@ initpyhbac(void)
                         "hbac.HbacError", HbacError__doc__,
                         PyExc_EnvironmentError, NULL);
     Py_INCREF(PyExc_HbacError);
-    ret = PyModule_AddObject(m, sss_py_const_p(char, "HbacError"), PyExc_HbacError);
-    if (ret == -1) MODINITERROR;
+    if (PyModule_AddObject(m, sss_py_const_p(char, "HbacError"),
+                           PyExc_HbacError) < 0) {
+        Py_DECREF(PyExc_HbacError);
+        Py_DECREF(m);
+        MODINITERROR;
+    }
 
     /* HBAC rule categories */
-    ret = PyModule_AddIntMacro(m, HBAC_CATEGORY_NULL);
-    if (ret == -1) MODINITERROR;
-    ret = PyModule_AddIntMacro(m, HBAC_CATEGORY_ALL);
-    if (ret == -1) MODINITERROR;
+    if (PyModule_AddIntMacro(m, HBAC_CATEGORY_NULL) < 0) {
+        Py_DECREF(PyExc_HbacError);
```

Comment:
I'm not sure this is required/valid.
At this point `m` should own `PyExc_HbacError` (see https://docs.python.org/3/c-api/module.html#c.PyModule_AddObject : `This steals a reference to value on success.`)